src/driver/hermes_cdev.c

```diff
@@ -157,8 +168,8 @@ static ssize_t hermes_write(struct file *filp, const char __user *buff,
 	chnl = xdma_get_h2c(hpdev);
 
 	pos = cfg->ehdsoff + *f_pos + env->data_slot * cfg->ehdssze;
-
-	res = xdma_channel_read_write(chnl, (char *) buff, count, &pos, true);
+	iov_iter_init(&iter, WRITE, &iovec, 1, count);
+	res = xdma_channel_read_write(chnl, &iter, pos, cfg->ehdssze - *f_pos);
```

Comment:
Why the max bytes? In this case it seems impossible for it to trigger. In other cases you could just use iov_iter_truncate().

Reply:
Because the iter might try to read/write beyond the program/data slot. iov_iter_truncate() is a good solution.